File: leetcode/209. Minimum Size Subarray Sum.cpp
class Solution {
public:
    int minSubArrayLen(int s, vector<int>& nums) {
        
        int n = nums.size();
        if(n == 0) return 0;
        int j,temp,minGroup = INT_MAX;
        
        for(int i = 0; i < n ; i++){
            j = i;
            temp = s;
            // cout << "  i ->" << i << endl;
            // cout << "TEMP ->" << temp << endl; 
            while(temp > 0  && j < n){
                temp -= nums[j];
                //cout << "temp ->" << temp << endl;
                j++;
            }
            //cout << "j ->" << j << endl;
            if( temp <= 0 && j - i < minGroup) minGroup = j - i;
            //cout << "------" << endl;
        }
        
        if(minGroup == INT_MAX) return 0;
        return minGroup;
        
    }
};

File: online-training-platform/leetcode/209. Minimum Size Subarray Sum.kt
class Solution {
    fun minSubArrayLen(target: Int, nums: IntArray): Int {
        
        var accum = 0; var start = 0; var end = 0
        var result = Int.MAX_VALUE
        
        while(end < nums.size){
            accum += nums[end]

            while(start < nums.size && accum - nums[start] >= target) {
                accum -= nums[start]
                start += 1
            }

            //println("$start - $end  | $accum")
            if(accum >= target){
                result = kotlin.math.min(result, end - start + 1)
            }
            end += 1
        }

        return if(result == Int.MAX_VALUE) 0 else result
    }
}